OpenRA.Mods.Common/Traits/Turreted.cs

```diff
@@ -255,9 +255,10 @@ public bool FaceTarget(Actor self, in Target target)
 			if (IsTraitDisabled || IsTraitPaused || attack == null || attack.IsTraitDisabled || attack.IsTraitPaused)
 				return false;
 
-			if (target.Type == TargetType.Invalid)
+			if (!target.IsValidFor(self))
 			{
-				desiredDirection = WVec.Zero;
+				realignDesired = true;
+				MoveTurret();
```

Comment:
This will probably break `RealignDelay`. Isn't setting `desiredDirection = WVec.Zero;` and letting `Tick` do the work enough?

Reply:
I've also had concerns, but it seems this is necessary for the unit to actually stop targeting. 
```C#
desiredDirection = WVec.Zero;
```
does nothing

Reply:
You are right, this is also not correct place for the fix

Changing `Target.Type != TargetType.Invalid` to `Target.IsValidFor(self)` everywhere in https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Mods.Common/Traits/Attack/AttackFollow.cs#L126-L148 seems to fix the bug and also respect `RealignDelay`

I'd not advise changing this every else though till #17227 gets implemented / addressed